src/packages/builder/src/codegen/index.ts

```diff
@@ -1,114 +1,113 @@
+	fs.writeFileSync('src/types.generated.ts', content.concat('\n', typesFileContent));
+	fs.writeFileSync('src/frontend/types.ts', content.concat('\n', typesFileContent));
+};
 
-		const typesOutputPaths = formatListOfTypeOutputPaths(options?.typesOutputPath);
+const generateIndividualTypeFiles = async (
+	config: CodegenConfig,
+	schema: DocumentNode,
+	documents: Source[]
+) => {
+	const argsList = await nearOperationFilePreset.buildGeneratesSection({
+		presetConfig: {
+			extension: '.generated.ts',
+			baseTypesPath: 'types.generated.ts',
+		},
+		plugins: [{ typescriptOperationsPlugin: {} }, { typescriptReactApolloPlugin: {} }],
+		schema,
+		documents,
+		config,
+		pluginMap: {
+			typescriptOperationsPlugin: typescriptOperations,
+			typescriptReactApolloPlugin: typescriptReactApollo,
+		},
+		baseOutputDir: '.',
+	});
 
-		// Write the files to disk
-		const writeOperations = files.flatMap((file) => [
-			fs.promises.writeFile(path.join(file.filename), file.content, 'utf8'),
-			// We save the types to two locations src and .graphweaver / outdir
-			...(file.filename === 'src/types.generated.ts'
-				? typesOutputPaths.map((outputPath: string) =>
-						fs.promises.writeFile(outputPath, file.content, 'utf8')
-				  )
-				: []),
-		]);
+	for (const args of argsList) {
+		const output = await codegen(args);
 
-		// Write the files to disk
-		await Promise.all(writeOperations);
-	} catch (err: any) {
-		const defaultStateMessage = `Unable to find any GraphQL type definitions for the following pointers:`;
-		if (err.message && err.message.includes(defaultStateMessage)) {
-			// do nothing for now and silently fail
-		} else {
-			console.log(err.message + `\n in ${err.source?.name}`);
-		}
+		fs.writeFileSync(args.filename, output);
 	}
+
+	const tasks = argsList.map(async (args) => {
+		const output = await codegen(args);
+		fs.promises.writeFile(args.filename, output);
+	});
+
+	await Promise.all(tasks);
 };
 
-const formatListOfTypeOutputPaths = (typesOutputPath?: string | string[]) => {
-	const typesOutput = ['./src/frontend/types.ts'];
+const generateFiles = async (schemaAsString: string) => {
+	const gqlSchema: GraphQLSchema = buildSchema(schemaAsString);
+	const schema = parse(printSchema(gqlSchema));
 
-	// If the typesOutputPath is a string or an array of strings, add it to the typesOutput array
-	if (typesOutputPath && typeof typesOutputPath === 'string') {
-		typesOutput.push(typesOutputPath);
-	}
-	if (typesOutputPath && Array.isArray(typesOutputPath)) {
-		typesOutput.push(...typesOutputPath);
+	const config = {
+		skipDocumentsValidation: {
+			skipDuplicateValidation: true, // A flag to disable the validation for duplicate query and mutation names we don't need this as we are using near-operation-file
+		},
+	};
+
+	let documents: Source[] | undefined = undefined;
+	try {
+		documents = await loadDocuments('./src/**/!(*.generated).{ts,tsx}', {
+			loaders: [new CodeFileLoader()],
+		});
+	} catch (e) {
+		documents = [];
```

Comment:
What kind of errors is this hiding? If I run types generation and it doesn't do anything because of this catch, wouldn't it be nice to know with at least a log?

Reply:
This code is actually from the CLI; the common error is that none of the files matching this pattern are relevant to codegen: `Unable to find any GraphQL type definitions for the following pointers`

I will add a log here.

Reply:
Logging added.